inc/packages/admin/namespace.php

```diff
@@ -518,3 +574,33 @@ function maybe_add_data_to_description( $description, $plugin ) {
 	$description .= '</p><p class="authors"><em>' . sprintf( __( 'Hosted on %1$s', 'fair' ), esc_html( $repo_host ) ) . '</em>';
 	return $description;
 }
+
+/**
+ * Filters the theme description when preparing themes for JS.
+ *
+ * @param array $themes Array of themes prepared for JS.
+ * @return array Array of themes with possible modifications.
+ */
+function maybe_add_data_to_theme_description( $themes ) {
+	foreach ( $themes as &$theme ) {
+		$did = get_file_data( get_stylesheet_directory() . '/style.css', [ 'ThemeID' => 'Theme ID' ] )['ThemeID'];
+		if ( empty( $did ) || ! str_starts_with( $did, 'did:plc:' ) ) {
+			continue;
+		}
+
+		$repo_host = Info\get_repository_hostname( $did );
+		if ( empty( $repo_host ) ) {
+			continue;
+		}
+
+		/* translators: %1$s: repository hostname */
+		$additional_description = '<p class="authors"><em>' . sprintf( __( 'Hosted on %1$s', 'fair' ), esc_html( $repo_host ) ) . '</em>';
+		if ( empty( $theme->description ) ) {
+			$theme['description'] .= '</p>' . $additional_description;
+		} else {
+			$theme['description'] .= $additional_description . '</p>';
+		}
+	}
+	unset( $theme );
+	return $themes;
```

Comment:
Do we need a new line before `return`?

Reply:
Done in [756f8ac](https://github.com/fairpm/fair-plugin/pull/284/commits/756f8ac10fd613d5ebb91ae8784747e6cd6bb1ae).